### PR TITLE
Statistical analysis

### PR DESCRIPTION
**Pull request by jonasleitner**
I'm not sure about the last task... at least i don't really know what should be plotted there. Maybe one of you has an idea :)

### CODE REVIEW

**Comment by omzoqasd** [comment]
hey guys,
i am currently on sick leave. hopefully can be back tomorrow

**Comment by jonasleitner** [comment]
then get well soon :) ... or however it is correct in english

**Comment by h-varma** [comment]
Get well soon @omzoqasd :)

**Comment by h-varma** [comment]
Everything looks good to me. I just have a couple of short comments/questions:
1. I fixed a minor bug: plt.plt.savefig -> plt.savefig
2. I updated the code to plot the euclidean distance, i.e., I assumed that there is a missing "...and so on" in the problem statement ;)
3. Did you skip task 4 because you didn't want to save some file in the folder?

**Comment by jonasleitner** [comment]
about the L2 norm: i think we both made a mistake and used the rows instead of columns :D
And about the plot... I just don't know what to plot then... because we have 3 numbers
But maybe I'm still missing something.

about task 4: ah yes, because i was lazy and didn't want to layout the output for creating a "nice" output file. And i thought, that we already have the result :D

**Comment by h-varma** [comment]
> about the L2 norm: i think we both made a mistake and used the rows instead of columns :D
> And about the plot... I just don't know what to plot then... because we have 3 numbers
> But maybe I'm still missing something.

Oh yes, you're right. Now I also checked the original file table.dat directly to see what's going on. I noticed that the original table has the headers r(x), v(x), r(y), v(y), r(z), r(z). Since these vectors represent physical quantities like dipole moments or something, it seems unlikely that we're actually working in a 4950-dimensional euclidean space. So maybe we just need to take the euclidean distance between each pair of columns for each row separately and plot them (see latest commit). 

What do you think?

> about task 4: ah yes, because i was lazy and didn't want to layout the output for creating a "nice" output file. And i thought, that we already have the result :D

Haha, okay then. All good :)

**Comment by jonasleitner** [comment]
> Since these vectors represent physical quantities like dipole moments or something, it seems unlikely that we're actually working in a 4950-dimensional euclidean space.

I actually think this may be the case. I think that the data is the flattened upper triangular of a Matrix... probably the dipole Matrix <a|mu|b>? (The first two columns in the file could be the indices) So I think we are comparing how much the Matrix elements are changing when switching the Gauge. (Whatever this means exactly...)

**Comment by h-varma** [comment]
> > Since these vectors represent physical quantities like dipole moments or something, it seems unlikely that we're actually working in a 4950-dimensional euclidean space.
> 
> I actually think this may be the case. I think that the data is the flattened upper triangular of a Matrix... probably the dipole Matrix <a|mu|b>? (The first two columns in the file could be the indices) So I think we are comparing how much the Matrix elements are changing when switching the Gauge. (Whatever this means exactly...)

If we are comparing matrix elements like you say (where we have 4950 matrix elements), doesn't this mean that we will get a vector containing 4950 components where each component is the distance between the corresponding matrix elements?

**Comment by jonasleitner** [comment]
If we would like to compare each matrix element: yes
But if we just want to compare how much the matrix as a whole is changing it would suffice to compare the Norm of (M1-M2) or in our case the norm of the difference of the 2 columns. (And yes there are different Matrix norms, so probably one has to think about for what kind of norm one can flatten the Matrix without changing the result. But probably you know that better than me :D)

**Comment by h-varma** [comment]
> If we would like to compare each matrix element: yes
> But if we just want to compare how much the matrix as a whole is changing it would suffice to compare the Norm of (M1-M2) or in our case the norm of the difference of the 2 columns. (And yes there are different Matrix norms, so probably one has to think about for what kind of norm one can flatten the Matrix without changing the result. But probably you know that better than me :D)

Hmm, matrix norms that take the norm element-wise rather than row-/column-wise should be unaffected by the flattening, right? In any case, I think that we might be seriously over-complicating the problem ;)

Let's boil it down to two options:

for two vectors r(x) and v(x) with dimensions (4950, 1)
1. Euclidean distance taken for each row independently -> || r_i(x) - v_i(x) ||_2 for i = 1, 2, ..., 4950
2. Euclidean distance between the two columns as a whole (~ Frobenius norm?) -> || r(x) - v(x) ||_F

Both could work depending upon what are actually looking for. Maybe @omzoqasd can pick one when he's back :)

**Comment by jonasleitner** [comment]
> I think that we might be seriously over-complicating the problem ;)
absolutely :D

**Review by omzoqasd — COMMENTED** [review]
nevermind

**Review by omzoqasd — COMMENTED** [review]
Nevermind. got it with the .var thingy

File: src/notebook_unit1.ipynb [hunk 1]
```diff
@@ -320,7 +323,11 @@
    "outputs": [],
    "source": [
     "# get rid of time column, lower triangular and diagonal entries of the correlation matrix\n",
-    "# sort the remaing values according to their absolute value, but keep the sign"
+    "# sort the remaing values according to their absolute value, but keep the sign\n",
+    "corr_mat = npop_filtered.drop(\"time\", axis=1).corr()\n",
+    "corr = corr_mat.where(np.triu(np.ones_like(corr_mat, dtype=bool), k=1)) #get rid of lower triangle\n",
+    "corr_vec = corr.unstack().dropna()\n",
+    "corr_vec.reindex(corr_vec.abs().sort_values(ascending=False).index) #sort according to absolute value"
```
Comment (omzoqasd) [inline src/notebook_unit1.ipynb]: @jonasleitner it solves the problem well. to be honest, I was wondering to get this code more readable and understandable. 
readable -> some comments?
understandable -> you do multiple tasks in line of code. that works great. maybe it is better for a better understanding to split it in small pieces step by step?

**Review by omzoqasd — COMMENTED** [review]
added comment here: commit a27c0ad38035c03e91e2a5cf2f505880d929777e

**Review by h-varma — APPROVED** [review]

**Pull request merged**